src/main/java/at/ac/tuwien/damap/conversion/AbstractTemplateExportScienceEuropeComponents.java

```diff
@@ -53,244 +53,273 @@ public void determinteDatasetIDs(){
+
+        }
+        // add grant number to funding item variables
+        if (project.getFunding() != null && project.getFunding().getGrantIdentifier().getIdentifier() != null) {
+            fundingItems.add(project.getFunding().getGrantIdentifier().getIdentifier());
+        }
+        // variable project funding, combination from funding item variables
+        if (!fundingItems.isEmpty()) {
+            addReplacement(replacements, "[grantid]", String.join(", ", fundingItems));
+        } else {
+            addReplacement(replacements, "[grantid]", "");
         }
     }
 
-    public void contributorInformation(){
-        //mapping contact information
-        if (dmp.getContact() != null) {
-            List<String> contactItems = new ArrayList<>();
-            String contactName = "";
-            String contactMail = "";
-            String contactId = "";
-            String contactIdentifierType = "";
-            String contactIdentifierId = "";
-            String contactAffiliation = "";
-            String contactAffiliationId = "";
-            String contactAffiliationIdentifierType = "";
-            String contactAffiliationIdentifierId = "";
-
-            if (dmp.getContact().getFirstName() != null && dmp.getContact().getLastName() != null) {
-                contactName = dmp.getContact().getFirstName() + " " + dmp.getContact().getLastName();
-                contactItems.add(contactName);
-            }
+    private String getContributorPersonIdentifier(Contributor contributor) {
+        String identifier = null;
+        Identifier personIdentifier = contributor.getPersonIdentifier();
 
-            if (dmp.getContact().getMbox() != null) {
-                contactMail = dmp.getContact().getMbox();
-                contactItems.add(contactMail);
+        if (personIdentifier != null) 
+        {
+            String contactIdentifierId = personIdentifier.getIdentifier();
+            if (personIdentifier.getIdentifierType().toString().equals("orcid")) {
+                identifier = "ORCID iD: " + contactIdentifierId;
             }
+        }
 
-            if (dmp.getContact().getPersonIdentifier() != null) {
-                contactIdentifierId = dmp.getContact().getPersonIdentifier().getIdentifier();
-                if (dmp.getContact().getPersonIdentifier().getIdentifierType().toString().equals("orcid")) {
-                    contactIdentifierType = "ORCID iD: ";
-                    contactId = contactIdentifierType + contactIdentifierId;
-                    contactItems.add(contactId);
-                }
-            }
+        return identifier;
+    }
 
-            if (dmp.getContact().getAffiliation() != null) {
-                contactAffiliation = dmp.getContact().getAffiliation();
-                contactItems.add(contactAffiliation);
-            }
+    private String getContributorAffiliationIdentifier(Contributor contributor) {
```

Comment:
This method could be refactored to be more readable like
```private String getContributorAffiliationIdentifier(Contributor contributor) {
        if (contributor.getAffiliationId() != null && contributor.getAffiliationId().getIdentifierType().toString().equals("ror")) {
            return = "ROR: " + contributor.getAffiliationId().getIdentifier();
        }

        return null;
    }```
or like this
```private String getContributorAffiliationIdentifier(Contributor contributor) {
	Identifier affiliationIdentifier = contributor.getAffiliationId();
        if (affiliationIdentifier != null && affiliationIdentifier.getIdentifierType().toString().equals("ror")) {
            return = "ROR: " + affiliationIdentifier.getIdentifier();
        }

        return null;
    }```
The same thing could be done with getContributorPersonIdentifier.

Reply:
Not a fan of the first proposal, as `getAffiliationId` is called multiple times. Storing it in a temporary variable makes it easier to access and read imo.

Reason behind the current implementation is to leave the possibility for extension of the method open, if we add support for other identifiers besids ROR.

Reply:
Thanks for clarifying